src/context/existingProject.tsx

```diff
@@ -0,0 +1,53 @@
+import React, {
+  createContext,
+  FC,
+  ReactNode,
+  useContext,
+  useState,
+} from "react";
+import { useQuery } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { fetchProject } from "api/projects";
+
+interface ContextProps {
+  projectExists: boolean;
+  isProjectLoading: boolean;
+  setProject: (project: string) => void;
+}
+
+const initialState: ContextProps = {
+  projectExists: false,
+  isProjectLoading: false,
+  setProject: () => undefined,
+};
+
+export const ExistingProjectContext = createContext<ContextProps>(initialState);
+
+interface ProviderProps {
+  children: ReactNode;
+}
+
+export const ExistingProjectProvider: FC<ProviderProps> = ({ children }) => {
```

Comment:
I propose to rename this hook to `useProject`. It reads the project from the location (i.e. calls useLocation directly) and uses the useQuery to load the project. It should return as value the project, isLoading and projectExists. We don't need state inside, because the project/data comes from useQuery and the projectName from the url.

Remove local state to store the project from all other components. And remove the location hook from most components (the ones that don't need other info from the url than projectName).